lfs/pointer.go

```diff
@@ -132,6 +133,9 @@ func DecodeFrom(reader io.Reader) (*Pointer, io.Reader, error) {
 	}
 
 	p, err := decodeKV(bytes.TrimSpace(buf))
+	if p != nil {
```

Comment:
I think in Go, it's conventional that if the error is non-nil, then the non-error arguments will be the zero value unless otherwise document.  This is almost always the case, and since Go doesn't have a nice, tagged type like Rust's `Result`, returning the zero value when we don't care is as good as anything else and is usually shortest.

However, I don't mind being a bit more explicit here, so I'll add a check.